src/game/self_play_cmd.rs

```diff
@@ -27,65 +30,87 @@ struct SelfPlayArgs {
     explore_factor: f32,
     #[clap(long, default_value = "1")]
     threads: u32,
+    #[clap(long, default_value = "100000")]
+    cache_size: usize,
 }
 
-pub trait ValueFunctionBuilder<Game: IGame>: Sync + Send {
-    fn new_value_func(&self, model_path: &String) -> Box<dyn ValueFunction<Game>>;
+pub trait IModelBuilder<Game: IGame>: Sync + Send {

```

Comment:
If if the interface accept model name, it is already specific for NN rather than any value function.
All this abstraction is clumsy, but i didnt find easier way in rust

Reply:
So maybe NNetworkBuilder? Whatever you prefer

Reply:
done